se_denseslam/include/se/continuous/volume_template.hpp

```diff
@@ -74,15 +74,15 @@ class VolumeTemplate {
       return _map_index->get(scaled_pos.x(), scaled_pos.y(), scaled_pos.z());
     }
 
-    value_type get(const Eigen::Vector3f & p) const {
+    value_type get(const Eigen::Vector3f& p) const {
       const float inverseVoxelSize = _size/_extent;
       const Eigen::Vector4i scaled_pos = (inverseVoxelSize * p.homogeneous()).cast<int>();
         return _map_index->get_fine(scaled_pos.x(),
                                     scaled_pos.y(),
                                     scaled_pos.z());
     }
 
-    value_type operator[](const Eigen::Vector3f p) const {
+    value_type operator[](const Eigen::Vector3i& p) const {
       return _map_index->get(p.x(), p.y(), p.z());
```

Comment:
I don't think this function is used anywhere and it's there mainly for historical reasons. What if we remove it altogether?

Reply:
I used it to create a dumb volume dump, that's how I found the bug. Indeed I think it is not used anywhere else. It seems like a rather simple way of accessing the data for a given voxel. There might be some value in keeping it, even if it is just for debugging purposes. Is the recommended way to access voxel data to use the DataHandler class?

Reply:
No DataHandler is pretty specific to the functor classes. I think at the moment the preferred way is the rather cumbersome procedure of fetching a Node/Block and then modify the data directly from there. I don't think there's a method to wrap that in the octree interface. We can make one, but the API should clearly handle the case in which the voxel does not exists (allocation on the fly would be dangerous but it's a possibility).

Reply:
In that case wouldn't `operator[]` still be useful for easy, read-only access to the voxel data? At least until a proper API is implemented.

Reply:
Maybe, I think we should properly unify the API sooner than later. Anyway, I'm going to merge this and then we can think about the API later.